Lib/test/test_ctypes/test_prototypes.py

```diff
@@ -72,6 +72,22 @@ def test_paramflags(self):
         self.assertEqual(func(None), None)
         self.assertEqual(func(input=None), None)
 
+    def test_invalid_paramflags(self):
+        proto = CFUNCTYPE(c_int, c_char_p)
+        with self.assertRaises(ValueError):
+            func = proto(("myprintf", testdll), ((1, "fmt"), (1, "arg1")))
+
+    def test_invalid_setattr_argtypes(self):
+        proto = CFUNCTYPE(c_int, c_char_p)
+        func = proto(("myprintf", testdll), ((1, "fmt"),))
+
+        self.assertRaisesRegex(TypeError, "_argtypes_ must be a sequence of types",
+                               setattr, func, "argtypes", 123)
```

Comment:
You should be able to use `assertRaisesRegex` as a context manager too:
```suggestion
        with self.assertRaisesRegex(TypeError,
                                    "_argtypes_ must be a sequence of types"):
            func.argtypes = 123
```
(same below)

(Most `ctypes` tests were written before this was possible, so they're a bit less readable than they could be now.)